app/src/main/java/ch/hikemate/app/ui/map/HikeDetailScreen.kt

```diff
@@ -186,47 +207,62 @@ fun HikeDetails(route: HikeRoute, isSaved: Boolean, date: Timestamp?) {
                 modifier = Modifier.padding(16.dp).weight(1f),
             ) {
               Text(
-                  text = route.name ?: "Unnamed Hike",
+                  text = route.name ?: stringResource(R.string.map_screen_hike_title_default),
                   style = MaterialTheme.typography.titleLarge,
                   textAlign = TextAlign.Left,
                   modifier = Modifier.testTag(TEST_TAG_HIKE_NAME))
-              AppropriatenessMessageWrapper(isSuitable = true)
+              AppropriatenessMessage(isSuitable = true)
             }
 
             // Bookmark icon that is filled if the hike is Saved
             val bookmark =
                 if (isSaved) R.drawable.bookmark_filled_blue else R.drawable.bookmark_no_fill
             Image(
                 painter = painterResource(bookmark),
-                contentDescription = null,
+                contentDescription =
+                    if (isSaved)
+                        stringResource(R.string.hike_detail_screen_bookmark_hint_on_isSaved_true)
+                    else stringResource(R.string.hike_detail_screen_bookmark_hint_on_isSaved_false),
                 modifier = Modifier.size(60.dp, 80.dp).testTag(TEST_TAG_BOOKMARK_ICON),
                 contentScale = ContentScale.FillBounds,
             )
           }
 
-          // Elevation box
-          Box(
+          ElevationGraph(
+              elevationData = listOf(150.0, 175.5, 200.3, 225.8, 210.0),
               modifier =
                   Modifier.fillMaxWidth()
                       .height(30.dp)
-                      .background(Color.Gray)
+                      .padding(4.dp)
+                      .background(color = MaterialTheme.colorScheme.surface)
```

Comment:
Change: don't use `.background` in the modifier of an elevation graph. If you want to set the fill/stroke color of an elevation graph, use the `styleProperties` parameter of the composable. It accepts a fill color, a stroke color, and a stroke width.

Would be nice if the graph was of the color of the display hike.

```kotlin
val color = getRandomColor()
// Use color where you used getRandomColor()
```

Pass the color to `HikeDetails` as a new parameter :

```kotlin
Color(hikeColor) // To convert it from an Int to a Color
```

Then :

```kotlin
ElevationGraph(
              elevationData = listOf(150.0, 175.5, 200.3, 225.8, 210.0),
              modifier =
                  Modifier.fillMaxWidth()
                      .height(30.dp)
                      .padding(4.dp)
                      .testTag(TEST_TAG_ELEVATION_GRAPH),
            styleProperties = ElevationGraphStyleProperties(
              fillColor = hikeColor
            )
```

Reply:
I've changed it. Is it better this way? I have left the the stroke color the onSurface color, since its difficult to view some of the colors generated by the random generator.

Reply:
I think you forgot to use the `hikeLineColor` variable you defined. Currently you do not use it for the graph, so the graph is still black.

Reply:
Yes exactly, I tried to get the graph to be the same color as the line on the map, however then the graph is sometimes very bright colors, which are difficult to see on the white background, which is why I left the color to be used only for the line on the map.

Reply:
Ohh right sorry I did not understand. I'll create a task for this in the backlog, the graph is working, which is the most important part.